drivers/modem/hl7800.c

```diff
@@ -4999,6 +5003,75 @@ static int setup_gprs_connection(char *access_point_name)
 	return send_at_cmd(NULL, cmd_string, MDM_CMD_SEND_TIMEOUT, 0, false);
 }
 
+static int set_bands(const char *bands, bool full_reboot)
+{
+	int ret;
+	char cmd[sizeof("AT+KBNDCFG=#,####################")];
+
+	ret = 0;
+
+	snprintk(cmd, sizeof(cmd), "AT+KBNDCFG=%d,%s", ictx.mdm_rat, bands);
+	ret = send_at_cmd(NULL, cmd, MDM_CMD_SEND_TIMEOUT, MDM_DEFAULT_AT_CMD_RETRIES, false);
+	if (ret < 0) {
+		goto done;
+	}
+
+	if (!full_reboot) {
+		ret = send_at_cmd(NULL, "AT+CFUN=1,1", MDM_CMD_SEND_TIMEOUT,
+				  MDM_DEFAULT_AT_CMD_RETRIES, false);
+		if (ret < 0) {
+			goto done;
+		}
+
+		ret = modem_boot_handler("LTE bands were just set");
+		if (ret < 0) {
+			goto done;
+		}
+	} else {
+		ret = modem_reset_and_configure();
+	}
+done:
+	return ret;
+}
+
+int32_t mdm_hl7800_set_bands(const char *bands)
+{
+	int ret, i;
```

Comment:
It would be cleaner to declare the `int i` straight in the for loop below, if it is not used outside the for loop